src/AbstractContainer.php

```
@@ -480,48 +484,47 @@ public function exchangeArray($input)
         $old            = $storage[$name];
         $storage[$name] = $input;
         if ($old instanceof ArrayObject) {
-            return $old->getArrayCopy();
+            /** @var array<TKey, TValue> $array */
+            $array = $old->getArrayCopy();
+            return $array;
         }
 
-        return $old;
+        /** @var array<TKey, TValue> $array */
+        $array = $old;
+        return $array;
     }
 
-    /** @inheritDoc */
-    public function getIterator()
+    /**
+     * Create a new iterator from an ArrayObject instance
+     */
+    public function getIterator(): Iterator
```

Why narrow the type to `Iterator`? what if `$container` is `IteratorAggregate`? `Traversable<TKey, TValue>` also aligns with `ArrayObject::getIterator()`

---

ArrayObject::getIterator() returns `Iterator` which while extending `Traversable` [triggered errors in a previous commit](https://github.com/laminas/laminas-session/actions/runs/16191740468/job/45708892458). I made a hotfix to match the signature to `ArrayObject`, but went back to returning `Traversable` now (albeit with the `#[\ReturnTypeWillChange]` attribute to suppress the error)

---

Ah, OK. Didn't realise it was narrowed in `ArrayObject` 👍